### ISSUE TITLE
PP Future breaks quickedit options  and saves undesired values ( in this case of RankMath, but others may be possible)

### ISSUE THREAD

**Issue by designzwang** [publishpress/PublishPress-Future#339]
Hi,

When PP Future is enabled, the quickedit options of other plugins are messed up. Screenshot links attached, better than a long explanation. A Javascript error for admin-edit.js $categories is undefined pops up.

Example:
the RankMath plugin adds additional Meta option tick fields to the quickedit  section of every post/page/.... in the post overview page at  wp-admin/edit.php?post_type=page ( see screenshots at snipboard , links below)
When PP is enabled and the editor opens the quickedit section of a post, all tick fields are enabled, regardless of their prior status, and then they get saved - thus overwriting the prior settings. In the case of RankMath, there are mutual exclusive fields like index and noindex, and with PP future active it is possible to tick (and save)  both values.

## Expected Behavior

Options in the quickedit section that are added by other plugins should be left alone.

## Current Behavior

Options of the Quickedit sections are messed up, all fields get ticked, and a javascript error appears ( see Screenshot )

## Steps to Reproduce (for bugs)

1. Enable RankMath and set some options for a post or page
2. Enable PP Future and go to the post or page overview
3. Click the QuickEdit link of the post or page from step1 and see that all options are ticked and a javascript error appears.

## Context

Screen shots : 

1) this is the quickedit section of a sample page when PPfuture is disabled. It shows the configuration of the page as  noindex, noarchive,nosnippet - everithing OK :   https://snipboard.io/NPpdcf.jpg

2) this is the same admin page with quickedit of the same page when PP Future is active: 
Suddenly all fields are ticked, even mutually exclusive ones.
 https://snipboard.io/hbcoFA.jpg

3) this is the JS error. It appears as soon as a quickedit link is clicked on the admin page.
https://snipboard.io/1eyprA.jpg

Regards
Ulrich

**Comment by andergmartins** [comment]
@designzwang Thanks for reporting the bug. We can't see the Robots Meta box in the quick edit panel using RankMath free. Sorry, it seems to be a Pro feature and we don't have an account for testing or debugging. 

We will need your help to debug the issue and send us more information, or to create a test site where we could replicate it, with permissions to upload a version of the plugin with debug instructions. On this case, you could email us the access information to help@publishpress.com.

**Comment by designzwang** [comment]
Hello @andergmartins ! Thanks for checking this issue. I would like to help out, but creating a publicly available instance of the site, outside of the infrastructure where we deploy automatically,  would be quite  a hassle. Since fixing this issue is also in the best interest of RankMAth, I will first contact them via our support access and ask for both a download link or package of the pro plugin and a dev license key for you. Will get back to you as soon as I hear from them. Regards, Ulrich

**Comment by andergmartins** [comment]
Sounds like a good plan. Looking for helping @designzwang. Thanks.

**Comment by designzwang** [comment]
Hi, I have received a Download URL from RankMath for the purpose of finding the bug, but that should not be in a public place ;)  I will forward it to your email address that is published in your gihub profile.

**Comment by designzwang** [comment]
Hi ,
any news on this? Will it be addressed in the next relase, @andergmartins ? Meanwhile we had to discover (painfully) that not only the tick fields are changed when using quickedit, but we also lose the content of the SEO text fields, and that is really bad since the content of these fields is not easy to reconstruct. So basically,  at this time our only workaround is to not use quickedit at all... 
Regards ,
Ulrich

**Comment by andergmartins** [comment]
Hi @designzwang 

Sorry for the delay and for the inconvenience the issue is causing.
We have a new beta package addressing the quick edit. Please, could you test it and check if the issue still exists?

[publishpress-future-3.1.4-beta.4.zip](https://github.com/publishpress/PublishPress-Future/files/13572803/publishpress-future-3.1.4-beta.4.zip)

**Comment by designzwang** [comment]
Hello @andergmartins ,
thanks for supplying a beta, however I did not have the time to test it. 

Now that we are on 3.1.5 ,  I tried it on my staging site, and it still is faulty. Look here:
![image](https://github.com/publishpress/PublishPress-Future/assets/5548014/bc42736b-b950-46f8-aefe-e8f642f9aaf9)

This was a newly created Post, that was created with noindex and nofollow attributes. When opening the quickedit accordion of this post, ALL options are checked, which is wrong. After saving in quickedit, all options are set. 
So we still have a situation where using quickedit is impossible since it alters those settings and especially it does not reflect the settings  of the post but activates all settings for robots meta.

As soon as PPFuture is disabled, this behaviour is fixed and the checkmarks reflect the correct settings of the posts/pages
Regards
Ulrich

Regards
Ulrich
